src/locales/i18n.ts

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import i18n from "i18next"
+import { initReactI18next } from "react-i18next"
+import homeIT from "./jsons/it/home.json"
+import homeEN from "./jsons/en/home.json"
+import commonIT from "./jsons/it/common.json"
+import commonEN from "./jsons/en/common.json"
+import FreeClassIT from "./jsons/it/freeClass.json"
+import FreeClassEN from "./jsons/en/freeClass.json"
+import settingsIT from "./jsons/it/settings.json"
+import settingsEN from "./jsons/en/settings.json"
+import { getLocales } from "expo-localization"
+
+void i18n.use(initReactI18next).init({
+  resources: {
+    it: {
+      common: commonIT,
+      home: homeIT,
+      freeClass: FreeClassIT,
+      settings: settingsIT,
+    },
+    en: {
+      common: commonEN,
+      home: homeEN,
+      freeClass: FreeClassEN,
+      settings: settingsEN,
+    },
+  },
+  lng: getLocales()[0].languageCode,
+  compatibilityJSON: "v3",
+  fallbackLng: "en",
+  defaultNS: "common",
+  ns: ["common", "home", "freeClass", "settings"],
+  interpolation: {
+    escapeValue: false,
+  },
+  react: { useSuspense: false },
+})
```

Comment:
qua direi di fare una cosa più carina dato che _tecnicamente_ l'inizializzazione di i18n è asincrona.
innanzitutto si può wrappare questa cosa in una funzione tipo
```ts
async function initializeI18n() {
    await i18n.use(...).init({...})
}
```

poi possiamo fare una cosa ancora più carina e creare un hook tipo 
```ts
export function useLoadI18n() {
  const [i18nLoaded, setI18nLoaded] = React.useState(false)

  React.useEffect(() => {
    void initializeI18n().then(() => setI18nLoaded(true))
  }, [])

  return i18nLoaded
}
```

e poi in App.tsx aggiungere questa cosa alla lista delle cose da inizializzare all'inizio (da aggiungere alla roba a riga 129)
```ts
import { useLoadI18n } from "./src/locales/i18n"
...

  const i18nInilitalized = useLoadI18n()

  useEffect(() => {
    if (settingsReady && fontsLoaded && tokensLoaded && i18nInilitalized) {
      void hideAsync().then(async () => {
        if (loginState.loggedIn) {
          console.log(await api.user.getPoliNetworkMe())
          console.log(await api.user.getPolimiUserInfo())
        }
      })
    }
  }, [settingsReady, fontsLoaded, tokensLoaded, i18nInilitalized])

  if (!settingsReady || !fontsLoaded || !tokensLoaded || !i18nInilitalized)
    return null
```

così aspettiamo di renderizzare la UI finché non viene caricato i18n

Reply:
giusto molto meglio vero, thanks!